### DISCUSSION TITLE
Using non-applicable PvP perks together with the "start when players are ready" server setting makes it impossible to cancel the selection

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#14892] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This is an issue which happens in the lobby menu and is related to the new perk system in PvP.
Here is a video demonstration: https://www.youtube.com/watch?v=yMFitBFlZLU
See reproduction steps for a detailed explanation.

### Reproduction steps

1. Host a server. You don't need other players to join the server in order to reproduce this bug.
2. Check the "Start when players are ready" server setting in the general settings tab.
3. Select the PvP sub vs sub game mode and pick a team.
4. Select a submarine with no large hardpoints, like the Barsuk.
5. Pick a large turret as one of your perks. Railgun, for example.
6. Check the "ready to start" checkbox.
7. The server will try to start a round, but since you picked a non-applicable perk, it will show you the perk compatibility warning menu with a large "Cancel" button, which normally lets you abort the start of the round and reselect perks.
8. Clicking the "Cancel" button after the automatic start will make the menu disappear and then reappear immediately, since the game tries to start the round again right after you exit the menu.
9. You are now stuck in an endless loop, and your only choice is to allow the round to start without reselecting your perks.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens outside single player or multiplayer game modes (e.g. game launches on startup, something broken in the main menu)

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in the upcoming hotfix patch. Closing discussion.
